044-cluster-wide-volume-usage-quota-management.md

```diff
@@ -0,0 +1,295 @@
+kafka 3.3 we can now obtain the total and usable (available) bytes per log dir as part of the DescribeLogDirsResponse.
+
+Note: if a single disk contains multiple log dirs, it will be described multiple times through the kafka APIs. This
+repetition is acceptable as our new limit types will be applied per-volume, so redundant volume descriptions don't
+impact the outcome.
+
+The Cluster Volume Source will use this API to discover volume information for the whole cluster.
+
+The performance cost of this API was [discussed](https://lists.apache.org/thread/11zyqqnyg1wgf4jdo6pvn7hn51g3vf8r) 
+upstream as part of the KIP, which should be low cost. The performance impact of polling is managed by making the poll
+interval configurable. The poll interval will be configurable using the existing `client.quota.callback.static.storage.check-interval` 
+property (default is 0 - disabled).
+
+Setting `client.quota.callback.static.storage.hard` or `client.quota.callback.static.storage.soft` would be incompatible
+with the cluster sourced volumes and fail configuration of the plugin.
+
+##### Admin Client Configuration
+
+To obtain log dir descriptions through the admin api we need to construct an admin client.
+
+If using cluster sourced volumes we require the admin client bootstrap to be configured using `client.quota.callback.kafka.admin.bootstrap.servers`
+
+Additional admin client configuration can be passed using the form `client.quota.callback.kafka.admin.${configuration}`.
+For example: `client.quota.callback.kafka.admin.security.protocol=SSL`
+
+The strimzi-kafka-operator would be responsible for configuring the admin client to connect to the
+replication listener of the broker.
+
+##### Limit Type Configuration
+
+When operating with cluster sourced volume data the existing limit types applied to the aggregate
+used bytes of all volumes would be meaningless. Nodes enter and exit the active set as part of normal operation.
+So we propose that the existing limiting method should be incompatible with cluster sourced volumes.
+
+Instead, we propose introducing new limit types applied on a per-volume basis. Meaning that there is a single value
+for each limit which we test against each volume. i.e. we do not support limiting based on a specific volume.
+
+We retain the existing concept of a soft limit (where throttling begins) and a hard limit (where we effectively
+stop message production) for all limit types.
+
+The only requirement is a single hard-limit configuration, if no soft-limit is configured we will infer that soft-limit==hard-limit.
+
+The different limit types can be combined, so you could combine a usedBytesExceeds soft limit with a availableRatioBelow
+hard limit.
+
+Defining multiple soft limits or multiple hard limits would be an invalid state.
+
+The limits we want are:
+1. throttle if [usedBytes](#volume) is greater-than-or-equal-to some threshold for any volume
```

Comment:
Again, do we really want this?

Reply:
Removed usedBytes limit type, we don't need it